gosu-core-api/src/main/gosu/gw/lang/enhancements/CoreIterableEnhancement.gsx

```diff
@@ -557,4 +558,25 @@ enhancement CoreIterableEnhancement<T> : java.lang.Iterable<T> {
     }
     return retList
   }
-}
\ No newline at end of file
+
+  /**
+   * takes two lists and returns a list of corresponding <code>gw.util.Pair</code>s.
+   * If one input list is short, excess elements of the longer list are discarded.
+   */
+  function zip<R>( other : Iterable<R>) : Iterable<Pair<T,R>> {
+    if (other == null) {
+      throw new NullPointerException("other should be non-null")
+    }
+
+    var zippedLength = this.Count < other.Count ? this.Count : other.Count
+    var zipped = new ArrayList<Pair<T,R>>(zippedLength)
+
```

Comment:
Lazy construction of elements:

Since you're given Iterables as input, IMO you can make no assumptions of whether they'll fit in memory. Returning an Iterator-based Iterable would be O(0) memory, and computation stays as O(n). Allocating a List, whether LinkedList or ArrayList, is O(n) memory.

Again, I'm just being pedantic. I think the List vs Iterable ship has already sailed since the others return List and Set.